src/main/java/bitbucketpullrequestbuilder/bitbucketpullrequestbuilder/bitbucket/ApiClient.java

```diff
@@ -82,23 +85,11 @@ public HttpClient getInstanceHttpClient() {
     }
 
     public List<Pullrequest> getPullRequests() {
-        try {
-            return parse(get(v2("/pullrequests/")), Pullrequest.Response.class).getPullrequests();
-        } catch(Exception e) {
-            logger.log(Level.WARNING, "invalid pull request response.", e);
-            e.printStackTrace();
-        }
-        return Collections.EMPTY_LIST;
+        return getAllValues(v2("/pullrequests/"), 50, Pullrequest.class);
     }
 
     public List<Pullrequest.Comment> getPullRequestComments(String commentOwnerName, String commentRepositoryName, String pullRequestId) {
-        try {
-            return parse(get(v1("/pullrequests/" + pullRequestId + "/comments")), new TypeReference<List<Pullrequest.Comment>>() {});
-        } catch(Exception e) {
-            logger.log(Level.WARNING, "invalid pull request response.", e);
-            e.printStackTrace();
-        }
-        return Collections.EMPTY_LIST;
+        return getAllValues(v2("/pullrequests/" + pullRequestId + "/comments"), 100, Pullrequest.Comment.class);
```

Comment:
Not your code, but the fact that this code manually constructs the URL is a bit of a shame, since the pullrequest object contains an embedded link to it that clients are expected to use over manual link construction.